utils.h

```diff
@@ -164,8 +164,8 @@ queue<string> exec_command(const string& command)
     while ( fgets( buff, sizeof buff, fp ) != NULL && !feof(fp) )
     {
         tmp_string.assign(buff);
-        while (tmp_string.size() > 0 &&
-               tmp_string[tmp_string.size() - 1] == '\n' ||
+        while ((tmp_string.size() > 0 &&
+               tmp_string[tmp_string.size() - 1] == '\n') ||
                tmp_string[tmp_string.size() - 1] == '\r') {
```

Comment:
This grouping is incorrect and hides a legitimate warning. It explicitly embraces an existing bug. See #40.